packages/fxa-content-server/app/scripts/lib/router.js

```diff
@@ -407,7 +407,19 @@ Router = Router.extend({
       })}`;
       this.navigateAway(settingsLink);
     },
-    'signin(/)': createViewHandler(SignInPasswordView),
+    'signin(/)': function () {
+      this.createReactOrBackboneViewHandler('signin', SignInPasswordView, {
+        ...Url.searchParams(this.window.location.search),
+
+        email: this.user.get('emailFromIndex'),
+        hasLinkedAccount: this.user.get('hasLinkedAccount'),
+        hasPassword: this.user.get('hasPassword'),
+      });
+      // clear out because users can be navigated to `/signin` in flows that aren't
+      // directly from email-first/index, and we want an `email` query param to take
+      // precedence but want to read from local storage otherwise
+      // this.user.set('emailFromIndex', '');
```

Comment:
Thought I needed this, then discovered it was something else. I need to circle back to test some edge cases to see if we need it.

Probably hold on calling stuff out like this in review for now, I'll have it cleaned up Monday. 🙏